aws/resource_aws_dlm_lifecycle_policy_test.go

```diff
@@ -0,0 +1,292 @@
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval_unit", "HOURS"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.times.0", "21:42"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.retain_rule.0.count", "10"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.tags_to_add.tf-acc-test-added", "full"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.target_tags.tf-acc-test", "full"),
+				),
+			},
+			{
+				Config: dlmLifecyclePolicyFullUpdateConfig(),
+				Check: resource.ComposeTestCheckFunc(
+					checkDlmLifecyclePolicyExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "description", "tf-acc-full-updated"),
+					resource.TestCheckResourceAttrSet(resourceName, "execution_role_arn"),
+					resource.TestCheckResourceAttr(resourceName, "state", "DISABLED"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.resource_types.0", "VOLUME"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.name", "tf-acc-full-updated"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval", "24"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.interval_unit", "HOURS"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.create_rule.0.times.0", "09:42"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.retain_rule.0.count", "100"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.schedule.0.tags_to_add.tf-acc-test-added", "full-updated"),
+					resource.TestCheckResourceAttr(resourceName, "policy_details.0.target_tags.tf-acc-test", "full-updated"),
+				),
+			},
+		},
+	})
+}
+
+func dlmLifecyclePolicyDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).dlmconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_dlm_lifecycle_policy" {
+			continue
+		}
+
+		input := dlm.GetLifecyclePolicyInput{
+			PolicyId: aws.String(rs.Primary.Attributes["name"]),
+		}
+
+		out, err := conn.GetLifecyclePolicy(&input)
+
+		if err != nil {
+			if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
+				return nil
+			}
+			return err
+		}
+
+		if out.Policy != nil {
+			return fmt.Errorf("DLM lifecycle policy still exists: %#v", out)
+		}
+	}
+
+	return nil
+}
+
+func checkDlmLifecyclePolicyExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		return nil
+	}
+}
+
+func dlmLifecyclePolicyBasicConfig() string {
+	return fmt.Sprint(`
+resource "aws_iam_role" "dlm_lifecycle_role" {
+  name = "tf-acc-basic-dlm-lifecycle-role"
```

Comment:
Nit: This would've been found with `resource.ParallelTest()` (had it existed back then!), but resource names like these need to be randomized (check out other testing with `rName` variables).